frontend/src/services/Controller.ts

```diff
@@ -129,10 +130,27 @@ function getEventHandlers() {
 
     dataReady: (result: boolean) => backend.set({ key: 'dataReady', value: result }),
 
-    environment: (result: ILookup) => backend.set({ key: 'environment', value: result }),
+    environment: (result: ILookup) => {
+      console.log('ENVIRONMENT')
+      console.log(result)
+      backend.set({ key: 'environment', value: result })
+      Analytics.Instance.setOS(result.os)
+      Analytics.Instance.setOsVersion(result.osVersion)
+      Analytics.Instance.setArch(result.arch)
+      Analytics.Instance.setManufacturerDetails(result.manufacturerDetails)
+      console.log('MANUFACTURER DETAILS')
+      console.log(result.manufacturerDetails)
+    },
 
     preferences: (result: IPreferences) => backend.set({ key: 'preferences', value: result }),
 
+    //Analytics
+    setOSInfo: (osInfo: IosInfo) => {
+      Analytics.Instance.setOS(osInfo.os)
+      Analytics.Instance.setOsVersion(osInfo.version)
+      Analytics.Instance.setArch(osInfo.arch)
+    },
+
     // User
     'signed-out': () => auth.signedOut(),
```

Comment:
I would have assumed you'd track things on the backend here. Why was it that you're not?

Reply:
some things had to be done in the frontend.  It was easier to implement in one place then doing it in both.  This particular code would be needed for all the frontend calls regardless.

Reply:
I think this is fine. It is very light weight, and already being passed to the frontend.